File: 851-goat-latin/goat-latin.rb
# @param {String} sentence
# @return {String}
def to_goat_latin(sentence)
    words = sentence.split(" ")
    index = 0
    while index < words.length do
        if words[index][0].downcase == "a" || words[index][0].downcase == "e" || words[index][0].downcase == "i" || words[index][0].downcase == "o" || words[index][0].downcase == "u"
            words[index] += 'ma'
        else
            first_letter = words[index][0]
            words[index].slice!(0)
            words[index] = words[index] + first_letter + "ma"
        end
        index += 1
    end
    count = 1
    index2 = 0
    while index2 < words.length do
        words[index2] += "a" * count
        count += 1
        index2 += 1
    end
    words.join(" ")
end

File: 851-goat-latin/goat-latin.py
class Solution(object):
    def toGoatLatin(self, sentence):
        vowels = ["a", "e", "i", "o", "u"]
        words = sentence.split()
        temp_array = []
        num_of_as = 1
        for word in words:
            if (word[0].lower() in vowels):
                word += "ma"
                word += "a" * num_of_as
                temp_array.append(word)
            else:
                first_letter = word[0]
                new_word = word[1:]
                new_word += word[0]
                new_word += "ma"
                new_word += "a" * num_of_as
                temp_array.append(new_word)
            num_of_as += 1
        return " ".join(temp_array)
        